### PR TITLE
Geirst/introduce bucket space concept in proton persistence engine

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review
@vekterli FYI

### CODE REVIEW

**Comment by toregge** [comment]
@geirst, please rebase

**Comment by geirst** [comment]
@toregge please review again

File: searchcore/src/apps/tests/persistenceconformance_test.cpp [hunk 1]
```diff
@@ -186,6 +186,7 @@ class DocumentDBFactory : public DummyDBOwner {
                                _queryLimiter,
                                _clock,
                                docType,
+                               document::BucketSpace::placeHolder(),
```
Comment (toregge) [inline searchcore/src/apps/tests/persistenceconformance_test.cpp]: please use storage::spi::test::makeBucketSpace() in tests.

File: searchcore/src/apps/tests/persistenceconformance_test.cpp [hunk 2]
```diff
@@ -316,7 +317,7 @@ class MyPersistenceEngine : public DocDBRepoHolder,
             LOG(info, "putHandler(%s)", itr->first.toString().c_str());
             IPersistenceHandler::SP proxy(
                     new PersistenceHandlerProxy(itr->second));
-            putHandler(itr->first, proxy);
+            putHandler(document::BucketSpace::placeHolder(), itr->first, proxy);
```
Comment (toregge) [inline searchcore/src/apps/tests/persistenceconformance_test.cpp]: Use itr->second->getBucketSpace() instead ?

File: searchcore/src/tests/proton/documentdb/documentdb_test.cpp [hunk 3]
```diff
@@ -114,6 +114,7 @@ Fixture::Fixture()
     mgr.forwardConfig(b);
     mgr.nextGeneration(0);
     _db.reset(new DocumentDB(".", mgr.getConfig(), "tcp/localhost:9014", _queryLimiter, _clock, DocTypeName("typea"),
+                             document::BucketSpace::placeHolder(),
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/documentdb_test.cpp]: please use storage::spi::test::makeBucketSpace() in tests.

File: searchcore/src/tests/proton/persistenceengine/persistenceengine_test.cpp [hunk 4]
```diff
@@ -432,8 +433,8 @@ struct SimpleFixture {
           engine(_owner, _writeFilter, -1, false),
           hset()
     {
-        engine.putHandler(DocTypeName(doc1->getType()), hset.phandler1);
-        engine.putHandler(DocTypeName(doc2->getType()), hset.phandler2);
+        engine.putHandler(BucketSpace::placeHolder(), DocTypeName(doc1->getType()), hset.phandler1);
+        engine.putHandler(BucketSpace::placeHolder(), DocTypeName(doc2->getType()), hset.phandler2);
```
Comment (toregge) [inline searchcore/src/tests/proton/persistenceengine/persistenceengine_test.cpp]: please use storage::spi::test::makeBucketSpace() in tests.

File: searchcore/src/tests/proton/docsummary/docsummary.cpp [hunk 5]
```diff
@@ -209,7 +209,8 @@ class DBContext : public DummyDBOwner
         _configMgr.nextGeneration(0);
         if (! FastOS_File::MakeDirectory((std::string("tmpdb/") + docTypeName).c_str())) { abort(); }
         _ddb.reset(new DocumentDB("tmpdb", _configMgr.getConfig(), "tcp/localhost:9013", _queryLimiter, _clock,
-                                  DocTypeName(docTypeName), *b->getProtonConfigSP(), *this, _summaryExecutor, _summaryExecutor,
+                                  DocTypeName(docTypeName), document::BucketSpace::placeHolder(),
```
Comment (toregge) [inline searchcore/src/tests/proton/docsummary/docsummary.cpp]: please use storage::spi::test::makeBucketSpace() in tests.

**Comment by geirst** [comment]
@toregge please review again

**Comment by toregge** [comment]
LGTM

**Pull request merged**